scipy/special/_basic.py

```diff
@@ -2209,6 +2215,11 @@ def comb(N, k, exact=False, repetition=False):
     if repetition:
         return comb(N + k - 1, k, exact)
     if exact:
+        if int(N) != N or int(k) != k:
+            warnings.warn(
+                "Non-integer arguments when exact is True are deprecated.",
```

Comment:
```suggestion
                "Non-integer arguments are deprecated when exact=True; "
                "note that arguments will be cast to integers (as before)",
```

I think some context is good here, especially as users may not have realized that they're getting wrong results up until now.